Android/app/src/main/java/app/intra/net/dns/DnsPacket.java

```diff
@@ -216,4 +220,55 @@ public short getQueryType() {
     }
     return addresses;
   }
+
+  public int getTTL() {
+    boolean hasTTL = false;
+    int minTTL = Integer.MAX_VALUE;
+    for (DnsRecord[] src : new DnsRecord[][]{answer, authority, additional}) {
```

Comment:
The code seems more conservative than Section 5.1 of the draft.  One difference is that the TTL only needs to be the minimum of the answer records, if there are answer records present.

Reply:
That section also mentions the SOA response, which appears in the Authority section for NXDOMAIN responses and sets their cache lifetime.  Although it references the SOA MINIMUM value, I think by the time the response gets to us the SOA TTL should also include that information (RFC 2308 Section 5).

The significance of the Additional section is less clear.  However, ultimately, the purpose of `getTTL` is to avoid the situation where `reduceTTL` has to clamp a negative TTL to zero, i.e. extend the lifetime of an RRset past its stated expiration.  I think it's clear that `reduceTTL` should be decrementing the TTLs of all RRsets in the message, regardless of section, so it follows that `getTTL` should account for them.